gmso/core/atom.py

```diff
@@ -0,0 +1,137 @@
+import warnings
+from typing import Optional, Union
+
+import unyt as u
+
+from pydantic import Field, validator
+
+from gmso.core.element import Element
+from gmso.abc.abstract_site import Site
+from gmso.core.atom_type import AtomType
+from gmso.utils._constants import UNIT_WARNING_STRING
+
+
+class Atom(Site):
+    __base_doc__ = """An atom represents a single element association in a topology.
+    
+    Atoms are the representation of an element within `gmso` that describes any general 
+    atom in a molecular simulation. Atoms also contain information that are unique to 
+    elements vs other types of interaction sites in molecular simulations. 
+    For example, charge, mass, and periodic table information.
+
+    Notes
+    -----
+    Atoms have all the attributes inherited from the base Site class
```

Comment:
Should add that "Order of precedence when attaining charge: atom.charge -> atom.atom_type.charge -> None"

Reply:
Same for mass